src/grid/grid3D.cpp

```diff
@@ -327,12 +334,12 @@ void Grid3D::AllocateMemory(void)
 
 
   #ifdef CHEMISTRY_GPU
-  C.HI_density    = &C.scalar[ 0*H.n_cells ];
-  C.HII_density   = &C.scalar[ 1*H.n_cells ];
-  C.HeI_density   = &C.scalar[ 2*H.n_cells ];
-  C.HeII_density  = &C.scalar[ 3*H.n_cells ];
-  C.HeIII_density = &C.scalar[ 4*H.n_cells ];
-  C.e_density     = &C.scalar[ 5*H.n_cells ];
+  C.HI_density     = &C.density[ H.n_cells*grid_enum::HI_density ];
```

Comment:
Would it be possible / make more sense to lines like this to reference C.host instead of C.density? I think one of the goals with this approach is to move away from having things point to arrays with confusing names, and since the creation of C.host and C.device, I don't actually know that we need the old conserved array pointers any more.

Reply:
Great point--agreed! How many of those changes do you want included in this PR specifically? Since changing all calls everywhere would be substantially more work.

FYI I started a discussions thread where I am listing potential to-dos (for a hacking day or otherwise) related to enum. 

https://github.com/cholla-hydro/cholla/discussions/215

Reply:
I think for this PR just using C.host or C.device where you've already changed them is fine. We can work on making it more consistent through the code base later.